src/render/posts/does-tdd-work.html.md

```diff
@@ -0,0 +1,228 @@
+
+* [TW Hangouts   |   Is TDD dead? - YouTube](https://www.youtube.com/watch?v=z9quxZsLcfo)
+* [TW Hangouts   |   Is TDD dead? Part II - YouTube](https://www.youtube.com/watch?v=JoTB2mcjU7w)
+* [TW Hangouts   |   Is TDD dead? Part III - YouTube](https://www.youtube.com/watch?v=YNw4baDz6WA)
+* [TW Hangouts   |   Is TDD dead? Part V & VI - YouTube](https://www.youtube.com/watch?v=gWD6REVeKW4)
+
+One interesting detail in those videos was the discussion around a pleasurable workflow and if there is a notable distinction in this area between a mind that prefers to write a test to begin solving a problem or a mind that prefers not to write a test until some form of solution for the problem has already been written.
+
+All this covers the input of only a few people in the industry and there are many more examples. I am pointing these out here because, despite the rich debate that took place in all these blog posts and discussions, no conclusive answers were arrived at. It seems rare that one side convinces the other. That may just be the nature of things but maybe we can explore it a little deeper.
+
+In nearly all the discussions there was a consensus that automated testing is important. Most disagreements seemed to be around the perception of TDD as a process and perhaps the granularity of automated tests too. 
+
+As with so many subjects in software development, a lot of the information we have available is based on opinions and anecdotes.  We hear about many stories from the perspective of the story teller and we try to build up a picture of what is right so we can try to apply that to our own situations. We generally just try things to see if they work. Some data would be useful. Is it just that software development is very difficult to measured in this way? When it comes to TDD at least some efforts have been made to gather data. 
+
+## Studies on TDD
+In 2003 a study called [An Initial Investigation of Test Driven Development in
+Industry](http://staff.unak.is/andy/MScTestingMaintenance/Homeworks/STMHeima7TestDrivenDevelopment.pdf) was conducted with ’24 professional pair programmers’ where one group developed with TDD and another group used a more conventional (at the time) design-develop-test-debug waterfall approach. That study also references an earlier German Study that was run with 19 graduate students that concluded that 
+
+>test-first manner neither leads to quicker development nor provides
+an increase in quality. However, the understandability of the
+program increases, measured in terms of proper reuse of existing
+interfaces. 
+
+There were a lot of limitations to that study which the 2003 study were trying to address. 
+
+According to the Georg and Williams paper, the TDD group produced code that passed 18% more functional black box test cases but took 16% more time for development.
+
+> A hypothesis of this research was that the TDD approach  
```

Comment:
This part seems to be formatted strangely, like a premature line-break or something.